Created the backend, as well as connected the form

torch2424 opened the pull request:
closes #7 
may close the issue #8 
may close the issue #13 

Firstly, let me apologize for closing multiple issues in one branch, the other two issues need some more testing, and could be confirmed by anyone willing to serve up the site and test things out :)

I created a little node.js backend to ping some services for us (Such as ifttt for google sheets, an undocumented slack api endpoint to auto invite emails, and the github api to automate org invintes), that we can access from our frontend form, using a little AngularJS service. This was tested working well (Would be nice to do some additional testing however).

This also does some various frontend design improvements, and a lot of them were mentioned by @davidnuon  in #13 , and this fixes a lot of those. They mostly dealt with responsiveness, but any additional testing, and confirmation would be nice to confirm closing the issue.

Screenshot (Chrome Inspected Nexus 5x): 

![screenshot-localhost 9000 2016-03-06 02-04-32](https://cloud.githubusercontent.com/assets/1448289/13553473/d148838a-e33f-11e5-8f80-192db1ffe933.png)

P.S this does contain some testing code, such as localhost backend urls and things, that i left in, simply for easier testing by anyone wanting to check this out. Please let me know if this a problem, and I can remove it

torch2424 committed:
Generated backend, Added an ifttt key check

as the title sates,doing this for secure ifttt keys.Also, git ignored node modules in the backend

torch2424 committed:
Finished up the google sheets route

Simply needs some testing using curl

torch2424 committed:
Improved things a bit more by making a keys.json

Made a keys.json to be parsed by the backend before deploying :)

torch2424 committed:
Added some more route logic

Added support for submitting to slack and google sheets, and seems to be working well!

torch2424 committed:
Falling asleep, almost done with mini backend

Simply need some slight hlp with api's, and need to be a bit more awake, tested this
working slightly

torch2424 committed:
Currently have IFTTT and Slack working

Currently have ifttt and slack working, need to response with our post response however

torch2424 committed:
Finished the github support!

Made seperate routes for every command, and works with a test org

torch2424 committed:
Created a basic Keys.json skeleton

As the title states

torch2424 committed:
Finished up nicer slack responses

As the title states, the backend now returns nice and understandable
slack responses

torch2424 committed:
Merge branch 'enhancement/#7-use-the-slack-api-to-onboard-members' of https://github.com/codeandcoffeelb/signup into enhancement/#7-use-the-slack-api-to-onboard-members

torch2424 committed:
Added nice ifttt responses

As the title states, simply need to finish the github responses

torch2424 committed:
Working on git response

@davidnuon shreked my computer with concentrated coffe spray, cant finish this branch for
now :'( :'( Y U DO DIS??!?!?! I CRIE EVERYTIM

torch2424 committed:
Finished the github responses

Working great!

torch2424 committed:
Made the keys.json error message more informative

torch2424 committed:
Got the backend and frontend linked together

Started to do so, and was testing successful responses with ifttt

torch2424 committed:
Improved some card desing and responsiveness

by requested by @davidnuon

torch2424 committed:
Fixed some other small front end things

Such as github user names for our requests and things

torch2424 committed:
More connecting the front and backend

Tested working with ifttt and slack, simply need to work out some more github weirdness

davidnuon commented:
Still reviewing the code, I can see one change that can be done. Please turn the coffee cup into an image.

Other than that, looking good so far!

![screen shot 2016-03-07 at 9 25 28 pm](https://cloud.githubusercontent.com/assets/477742/13592897/3def2f4a-e4ab-11e5-8e2c-878ca4caac29.png)

davidnuon commented on backend/routes/index.js:
```diff
@@ -0,0 +1,174 @@
+
+/* GET home page. */
+router.get('/', function(req, res, next) {
+  res.render('index', { title: 'I am a Hamburger' });
```
lol

davidnuon replied:
![nom nom nom](http://vignette1.wikia.nocookie.net/yugioh/images/e/ec/HungryBurger-TF04-JP-VG.jpg/revision/latest?cb=20120611230756)

torch2424 replied:
What does that mean? Like I said I am a terrible backend guy, so any advice on what I did wrong would be cool! :)

rogerhoward replied:
I think he's just cracking up at the throwaway response to someone accessing /

Personally, I like returning HTTP 418 in this scenario :)

torch2424 replied:
OHHHHHHH I forgot I put that in there. Yeah julian and I have a joke, similar to googl'es teapot:

http://www.error418.net/

And we can definitely add a return of 418 haha

davidnuon commented:
Another kink in the animation in Chrome on OSX 
![coffee_kink](https://cloud.githubusercontent.com/assets/477742/13593156/c7b2d55e-e4ad-11e5-90fd-061ea91aba0f.gif)

torch2424 committed:
Allowed machines to access grunt serve

torch2424 committed:
David's Fixes

I fixed bugs that the awesome @davidnuon helped me find, Also, made some really
good transparent code and coffee logos, as well as some that are not transparent.
Lastly, also did some animation work with the divs on mobile, which were calling
some functions to fast, so I slowed the timeouts, as well as set a slightly larger
height on load, to keep users from having completely broken pages.

torch2424 commented:
I made your changes, thank you for the help! @davidnuon

torch2424 commented:
And it is quite weird OSX chrome was behaving like this, I tested on my chrome in linux, and on android and had no problems...

davidnuon commented:
All good :)
